pkg/queue/concurrency_state.go

```diff
@@ -58,15 +53,23 @@ func ConcurrencyStateHandler(logger *zap.SugaredLogger, h http.Handler, pause, r
 			case <-doneCh:
 				inFlight--
 				if inFlight == 0 {
-					logger.Info("Requests dropped to zero ...")
-					pause()
+					logger.Info("Requests dropped to zero")
+					if err := pause(endpoint); err != nil {
+						logger.Errorf("Error handling pause request: %v", err)
+						panic(err)
```

Comment:
Does this panic crash the entire pod?

Reply:
Based on some local testing, I believe it only kills the queue-proxy container. The user-container container doesn't seem to be impacted. Here's some kubectl output when a panic occurs:

```
$ k get po -w
NAME                                        READY   STATUS    RESTARTS   AGE
example-00001-deployment-598f4f7b76-7rngf   2/2     Running   0          107s
example-00001-deployment-598f4f7b76-7rngf   1/2     Error     0          111s
example-00001-deployment-598f4f7b76-7rngf   1/2     Running   1          112s
example-00001-deployment-598f4f7b76-7rngf   1/2     Running   1          113s
example-00001-deployment-598f4f7b76-7rngf   2/2     Running   1          114s
```

Reply:
Yeah, that's fine. I guess we'll want the entire pod to be recreated if that happens?

Reply:
Yes. Right now, the panic is just a place holder (see @julz 's comment [here](https://github.com/knative/serving/pull/11802#discussion_r692089319), but we'll be implementing something more robust shortly via #11834